WordPress/gutenberg
# Adding unit tests for `useCompositeState` to `Composite` component

## What?

This PR adds unit tests to validate `useCompositeState` behaviour.


## Why?

In order to ease [the migration away from `reakit`](https://github.com/WordPress/gutenberg/issues/53278), we need to maintain the current public `Composite` API. `reakit` exposes a `useCompositeState` function, which `ariakit` doesn't directly support, so [we need to port this](https://github.com/WordPress/gutenberg/issues/56548) to [the new `Composite` implementation](https://github.com/WordPress/gutenberg/issues/55224).

To ensure this work maintains existing behaviour, we need to have unit tests in place.


## How?

Tests are written against each prop and action provided by Reakit's `useCompositeState` implementation.

Additionally, tests are written for common use-cases to validate interaction and developer expectation.


## Testing Instructions

- Confirm that test coverage is complete
- Ensure that all tests run
